grakn-graph/src/main/java/ai/grakn/graph/internal/InstanceImpl.java

```diff
@@ -88,26 +92,20 @@
      */
     @Override
     public void delete() {
-        InstanceImpl<?, ?> parent = this;
-        Set<CastingImpl> castings = parent.castings();
+        Set<Relation> relations = castingsInstance().map(Casting::getRelation).collect(Collectors.toSet());
```

Comment:
Oh, I see. So some other method that `delete` calls is iterating over the relations?